content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,221 +11,289 @@ weight: 10
+* `None` - This volume mount will not receive any subsequent mounts
+  that are mounted to this volume or any of its subdirectories by the host.
+  In similar fashion, no mounts created by the container will be visible on
+  the host. This is the default mode.
 
-### cephfs
+  This mode is equal to `private` mount propagation as described in the
+  [Linux kernel documentation](https://www.kernel.org/doc/Documentation/filesystems/sharedsubtree.txt)
 
-A `cephfs` volume allows an existing CephFS volume to be
-mounted into your Pod. Unlike `emptyDir`, which is erased when a pod is
-removed, the contents of a `cephfs` volume are preserved and the volume is merely
-unmounted. This means that a `cephfs` volume can be pre-populated with data, and
-that data can be shared between pods. The `cephfs` volume can be mounted by multiple
-writers simultaneously.
+* `HostToContainer` - This volume mount will receive all subsequent mounts
+  that are mounted to this volume or any of its subdirectories.
 
-{{< note >}}
-You must have your own Ceph server running with the share exported before you can use it.
-{{< /note >}}
+  In other words, if the host mounts anything inside the volume mount, the
+  container will see it mounted there.
 
-See the [CephFS example](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/volumes/cephfs/) for more details.
+  Similarly, if any pod with `Bidirectional` mount propagation to the same
+  volume mounts anything there, the container with `HostToContainer` mount
+  propagation will see it.
 
-### cinder
+  This mode is equal to `rslave` mount propagation as described in the
+  [Linux kernel documentation](https://www.kernel.org/doc/Documentation/filesystems/sharedsubtree.txt)
 
-{{< note >}}
-Kubernetes must be configured with the OpenStack cloud provider.
-{{< /note >}}
+* `Bidirectional` - This volume mount behaves the same the `HostToContainer` mount.
+  In addition, all volume mounts created by the container will be propagated
+  back to the host and to all containers of all pods that use the same volume.
 
-The `cinder` volume type is used to mount the OpenStack Cinder volume into your pod.
+  A typical use case for this mode is a pod with a FlexVolume or CSI driver or
+  a pod that needs to mount something on the host using a `hostPath` volume.
 
-#### Cinder volume configuration example
+  This mode is equal to `rshared` mount propagation as described in the
+  [Linux kernel documentation](https://www.kernel.org/doc/Documentation/filesystems/sharedsubtree.txt)
+
+  {{< warning >}}
+  `Bidirectional` mount propagation can be dangerous. It can damage
+  the host operating system and therefore it is allowed only in privileged  
+  containers. Familiarity with Linux kernel behavior is strongly recommend  ed.
+  In addition, any volume mounts created by containers in pods must be destroyed
+  (unmounted) by the Containers on termination.
+  {{< /warning >}}
+
+### Mount Propagation Configuration
```

Comment:
```suggestion
### Mount propagation configuration
```
Could this move to a new task page? If someone is new to Kubernetes, they are going to see a lot of _very_ detailed information before they have even understood what Kubernetes means by a new volume. I'm genuinely worried about driving readers away if there's too much detail too early.

Reply:
What do you think about mount propagation as a separate concept page under storage?

I think the config section is too small to survive on it's own under tasks.